ext/ffi/jit_trampoline.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018-2022 the Deno authors. All rights reserved. MIT license.
+
+use crate::NativeType;
+use crate::{tcc::Context, Symbol};
+use std::ffi::c_void;
+use std::ffi::CString;
+use std::fmt::Write as _;
+
+pub(crate) struct Allocation {
+  pub addr: *mut c_void,
+  _ctx: Context,
+  _sym: Box<Symbol>,
+}
+
+macro_rules! cstr {
+  ($st:expr) => {
+    &CString::new($st).unwrap()
+  };
+}
+
+fn native_to_c(ty: &NativeType) -> &'static str {
+  match ty {
+    NativeType::U8 | NativeType::U16 | NativeType::U32 => "unsigned int",
+    NativeType::I8 | NativeType::I16 | NativeType::I32 => "int",
```

Comment:
I think we should extend support for 64bit `typeof arg === "number"` parameters. The Fast API supports those, and we support those in our normal FFI calls; we'll always return BigInts but we do allow users to pass in a number as a u64 / i64 / usize / isize. They're forced into only up-to 53bit values, but that's what they have in the JS numbers originally anyway so there's no risk of losing data or anything like that.

And if they need the full 64bits to represent the number, they'll use a BigInt which means that V8 will choose to not call the Fast API function.

Reply:
Sure, let's do this in a follow up. I want to keep this patch as a 'fix'.